app/src/main/java/org/schabi/newpipe/RouterActivity.java

```diff
@@ -425,87 +466,61 @@ private void showDialog(final List<AdapterChoiceItem> choices) {
 
     private List<AdapterChoiceItem> getChoicesForService(final StreamingService service,
                                                          final LinkType linkType) {
-        final Context context = getThemeWrapperContext();
-
-        final List<AdapterChoiceItem> returnList = new ArrayList<>();
-        final List<StreamingService.ServiceInfo.MediaCapability> capabilities
-                = service.getServiceInfo().getMediaCapabilities();
-
-        final SharedPreferences preferences = PreferenceManager
-                .getDefaultSharedPreferences(this);
-        final boolean isExtVideoEnabled = preferences.getBoolean(
-                getString(R.string.use_external_video_player_key), false);
-        final boolean isExtAudioEnabled = preferences.getBoolean(
-                getString(R.string.use_external_audio_player_key), false);
-
-        final AdapterChoiceItem videoPlayer = new AdapterChoiceItem(
-                getString(R.string.video_player_key), getString(R.string.video_player),
-                R.drawable.ic_play_arrow);
         final AdapterChoiceItem showInfo = new AdapterChoiceItem(
                 getString(R.string.show_info_key), getString(R.string.show_info),
                 R.drawable.ic_info_outline);
-        final AdapterChoiceItem popupPlayer = new AdapterChoiceItem(
-                getString(R.string.popup_player_key), getString(R.string.popup_player),
-                R.drawable.ic_picture_in_picture);
+        final AdapterChoiceItem videoPlayer = new AdapterChoiceItem(
+                getString(R.string.video_player_key), getString(R.string.video_player),
+                R.drawable.ic_play_arrow);
         final AdapterChoiceItem backgroundPlayer = new AdapterChoiceItem(
                 getString(R.string.background_player_key), getString(R.string.background_player),
                 R.drawable.ic_headset);
-        final AdapterChoiceItem addToPlaylist = new AdapterChoiceItem(
-                getString(R.string.add_to_playlist_key), getString(R.string.add_to_playlist),
-                R.drawable.ic_add);
+        final AdapterChoiceItem popupPlayer = new AdapterChoiceItem(
+                getString(R.string.popup_player_key), getString(R.string.popup_player),
+                R.drawable.ic_picture_in_picture);
 
+        final List<AdapterChoiceItem> returnedItems = new ArrayList<>();
+        returnedItems.add(showInfo); // Always present
```

Comment:
Uhm, why is it always present? Shouldn't it be hidden in cases where it would do the exact same thing as `videoPlayer`, to save some space?

Reply:
1. You would end up with an action "Video player" that behaves like "Show Info". If thought logically it should be exactly opposite: We shouldn't show "Video player" when it can be done with "Show info".
2. This would require a ton of if logic and another logical layer on top of the choices. Which is in my opinion overkill.
3. It may be unexpected for the user if no "Show info" or "Video player" is present.
4. Saving space is not relevant because there is still a huge portion of users that have all entries anyway visible (e.g. if the have auto-play enabled - which is default).

Reply:
Mmmh ok, then in a separate PR the router activity should be made scrollable. When I have the phone in landscape I can't see all options.

Reply:
Oh, I didn't notice that.
Sound's like a new issue 😉